File: src/main/java/frc/robot/commands/hatchpanel/ManualStingerControl.java
package frc.robot.commands.hatchpanel;

import edu.wpi.first.wpilibj.command.Command;
import frc.robot.Robot;
import frc.robot.subsystems.Stinger;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class ManualStingerControl extends Command {
    private final Logger LOG = LogManager.getLogger(ManualStingerControl.class);

    public ManualStingerControl() {
        requires(Robot.stinger);
        setName("ManualStingerControl Command");
        LOG.info(getName() + " Constructed");
    }

    @Override
    protected void initialize() {
    }

    @Override
    protected void execute() {
        //Robot.ejector.getEjector().setSpeed(Robot.oi.getGamepadAxis(RobotMap.GAMEPAD_AXIS.leftY));
        Robot.stinger.setState(Stinger.StingerState.EJECTING);
        LOG.info(getName() + " Executed");
    }

    @Override
    protected boolean isFinished() {
        return false;
    }

    @Override
    protected void end() {
        Robot.stinger.setState(Stinger.StingerState.IDLE);
        LOG.info(getName() + " Ended");
    }

    @Override
    protected void interrupted() {
        LOG.info(getName() + " Interrupted");
        this.end();
    }
}

File: src/main/kotlin/ktfrc/robot/commands/hatchpanel/ManualStingerControl.kt
package ktfrc.robot.commands.hatchpanel

import edu.wpi.first.wpilibj.command.Command
import ktfrc.robot.Robot
import ktfrc.robot.RobotMap
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

class ManualStingerControl: Command() {

    init {
        requires(Robot.stinger)
        name = "ManualStingerControl Command"
        LOG?.info("$name Constructed")
    }

    override fun execute() {
        Robot.stinger.ejector.setSpeed(Robot.oi.getGamepadAxis(RobotMap.GAMEPAD_AXIS.leftY))
        LOG?.info("$name Executed")
    }

    override fun isFinished(): Boolean {
        return false
    }

    override fun end() {
        Robot.stinger.ejector.stop()
        LOG?.info("$name Ended")
    }

    override fun interrupted() {
        LOG?.info("$name Interrupted")
        this.end()
    }

    companion object {
        private val LOG: Logger? = LogManager.getLogger(ManualStingerControl::class.java)
    }
}